pkg/awsutils/awsutils.go

```diff
@@ -838,33 +852,111 @@ func (cache *EC2InstanceMetadataCache) createENI(useCustomCfg bool, sg []*string
+		if err == nil {
+			log.Infof("Created a new ENI: %s", aws.StringValue(result.NetworkInterface.NetworkInterfaceId))
+			return *result.NetworkInterface.NetworkInterfaceId, nil
 		}
-		input.SubnetId = aws.String(subnet)
+		checkAPIErrorAndBroadcastEvent(err, "ec2:CreateNetworkInterface")
+		awsAPIErrInc("CreateNetworkInterface", err)
+		prometheusmetrics.Ec2ApiErr.WithLabelValues("CreateNetworkInterface").Inc()
+		log.Errorf("Failed to CreateNetworkInterface %v for subnet %s", err, customSubnet)
 	} else {
-		log.Info("Using same config as the primary interface for the new ENI")
+		subnetResult, vpcErr := cache.getVpcSubnets()
+		if vpcErr != nil {
+			return "", errors.Wrap(vpcErr, "failed to get subnets in vpc")
+		}
+
+		for _, subnet := range subnetResult {
+			if *subnet.SubnetId != cache.subnetID {
+				if !validTag(subnet) {
+					continue
+				}
+			}
+
+			log.Infof("Creating ENI with security groups: %v in subnet: %s", aws.StringValueSlice(input.Groups), aws.StringValue(input.SubnetId))
+
+			input.SubnetId = subnet.SubnetId
+
+			start := time.Now()
+			result, err = cache.ec2SVC.CreateNetworkInterfaceWithContext(context.Background(), input)
+			prometheusmetrics.Ec2ApiReq.WithLabelValues("CreateNetworkInterface").Inc()
+			prometheusmetrics.AwsAPILatency.WithLabelValues("CreateNetworkInterface", fmt.Sprint(err != nil), awsReqStatus(err)).Observe(msSince(start))
+			if err == nil {
+				log.Infof("Created a new ENI: %s", aws.StringValue(result.NetworkInterface.NetworkInterfaceId))
+				return *result.NetworkInterface.NetworkInterfaceId, nil
+			}
+			checkAPIErrorAndBroadcastEvent(err, "ec2:CreateNetworkInterface")
+			awsAPIErrInc("CreateNetworkInterface", err)
+			prometheusmetrics.Ec2ApiErr.WithLabelValues("CreateNetworkInterface").Inc()
+			log.Errorf("Failed to CreateNetworkInterface %v for subnet %s", err, *subnet.SubnetId)
+		}
 	}
+	return "", errors.Wrap(err, "failed to create network interface")
+}
 
-	log.Infof("Creating ENI with security groups: %v in subnet: %s", aws.StringValueSlice(input.Groups), aws.StringValue(input.SubnetId))
+func (cache *EC2InstanceMetadataCache) getVpcSubnets() ([]*ec2.Subnet, error) {
+	describeSubnetInput := &ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{aws.String(cache.vpcID)},
+			},
+			{
+				Name:   aws.String("availability-zone"),
+				Values: []*string{aws.String(cache.availabilityZone)},
+			},
+		},
+	}
 
 	start := time.Now()
-	result, err := cache.ec2SVC.CreateNetworkInterfaceWithContext(context.Background(), input)
-	prometheusmetrics.Ec2ApiReq.WithLabelValues("CreateNetworkInterface").Inc()
-	prometheusmetrics.AwsAPILatency.WithLabelValues("CreateNetworkInterface", fmt.Sprint(err != nil), awsReqStatus(err)).Observe(msSince(start))
+	subnetResult, err := cache.ec2SVC.DescribeSubnetsWithContext(context.Background(), describeSubnetInput)
```

Comment:
nit: we should investigate whether a paginated call would be better looks. Looks like this is unpaginated by default